book/src/concepts/entity_and_component.md

```diff
@@ -0,0 +1,105 @@
+
+## Entity and Component in Amethyst
+In an inheritance design, entity usually contains components. All the data and methods related to an entity are stored within. However, in the ECS design, entity is just a general purpose object. In fact, the implementation of `Entity` in Amethyst is simply:
+```rust,no_run,noplaypen,ignore
+struct Entity(u32, Generation);
+```
+where u32 is the id of the entity and generation is used to check if the entity has been deleted.
+
+`Entity`s are stored in a special container `EntitiesRes`. Whereas the data associated with the entities are grouped into components and stored in the designated storages.
+
+Consider an example where you have three objects: two bottles and a person.
+
+* Bottle A: x: 150.0, y: 202.1, shape: "round", color: "red"
+
+* Bottle B: x: 15.0, y: 22.0, shape: "square", color: "white"
+
+* Person C: x: 100.5, y: 300.8, name: "Peter"
+
+We can separate bottle's properties into `PositionComponent` and `BottleComponent`, and person's properties into `PositionComponent` and `PersonComponent`. Here's an illustration of how the three objects would be stored.
+
+![How entity and components are stored](../images/concepts/component_and_entity.svg)
+
+As you could see from the graph, entities do not store data. Nor do they know any information about their components. They serve the purpose of object identification and tracking object existance. The components store all the data and their connection to entities. 
+
+If you know database systems, this organization looks quite similar to the tables in a database, where entity id serves as the key in each table. In fact, you can even join components or entities like joining tables. For example, to update the position of all the persons, you will need to join the PersonComponent and the PositionComponent. 
+
+## EntitiesRes
+
+Even though the structure of the entity is pretty simple, entity manipulation is very sophisticated and crucial to game performance. This is why entities are handled exclusively by the struct `EntitiesRes`. `EntitiesRes` provides two ways for creating/deleting entities:
+
+* Immediate creation/deletion, used for game setup or clean up.
+* Atomic or delayed creation/deletion, used in the game play state. It updates entities in batch at the end of each game loop.
+
+You will see how these methods are used in later chapters.
+
+
+## Declaring a component
+
+To declare a component, you first declare the relevant underlying data:
+
+```rust,no_run,noplaypen
+/// This `Component` describes the shape of an `Entity`
+enum Shape {
+    Sphere { radius: f32 },
+    RectangularPrism { height: f32, width: f32, depth: f32 },
+}
+
+/// This `Component` describes the transform of an `Entity`
+pub struct Transform {
+    position: Translation3<f32>,
+    rotation: UnitQuaternion<f32>,
+    scale: Vector3<f32>,
```

Comment:
Btw, the `Transform` type was updated in the last release, this would be consistent with the source in `master`:

```rust
#[derive(Clone, Debug, PartialEq)]
pub struct Transform {
    /// Translation + rotation value
    iso: Isometry3<f32>,
    /// Scale vector
    scale: Vector3<f32>,
}
```

Reply:
I used `position` `rotation` and `scale` to make it easier to understand (I had to look up what isometry3 was). But using the source code with comments feels like a better solution. I changed it :)

Reply:
cool thanks!